### PR TITLE
DOC: Epochs.next and Epochs.iter_evoked

### LINKED ISSUE #3099: Epochs.next() docs don't match behavior
[Docstring says here](https://github.com/mne-tools/mne-python/blob/master/mne/epochs.py#L1423-L1424) that `Epochs.next()` returns an instance of `mne.epochs.Epochs`, but in fact it returns a `numpy.ndarray` (or a tuple of `ndarray, event_id` if `return_event_id=True`).  Personally I think the docs reflect how it _should_ work, but it's obviously easier to just change the docs to reflect what it _actually_ does.

### PR DESCRIPTION
**Pull request by drammock**
closes #3099 . ready for review @agramfort

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -583,7 +583,7 @@ def _detrend_offset_decim(self, epoch, verbose=None):
         return epoch
 
     def iter_evoked(self):
-        """Iterate over Evoked objects with nave=1
+        """Iterate epochs as a sequence of Evoked objects with n_epochs=1
```
Comment (jasmainak) [inline mne/epochs.py]: `nave` is an attribute of `evoked` object. I feel you should keep that but add in brackets that it means only one epoch has been averaged ...

Comment (drammock) [inline mne/epochs.py]: True.  But `nave` is, IMO, simply not clear:

- It is not a parameter of the `Evoked` constructor.
- It does not follow a convention seen elsewhere in the code and docs of using `n_` as a prefix for something that is a count of items (n_samples, n_epochs, n_components, n_tapers, n_rows, n_cols, n_vertices, n_times...).
- In its `__repr__` method [it prints `nave` as `n_epochs`](https://github.com/mne-tools/mne-python/blob/master/mne/evoked.py#L126).  And docstrings, like `__repr__` output, are for humans to read.
- It is [an extant (though somewhat rare) word of English](http://www.merriam-webster.com/dictionary/nave), and might easily be mistaken for obscure neuroimaging term.

Comment (Eric89GXL) [inline mne/epochs.py]: How about something like:
```
"""Iterate over epochs as a sequence of Evoked objects

The Evoked objects will each contain a single epoch (i.e. no averaging is performed).
...
```
Thus we can avoid `nave` and `n_epochs` altogether?

Comment (drammock) [inline mne/epochs.py]: that's fine with me.  @jasmainak WDYT?

Comment (agramfort) [inline mne/epochs.py]: approved

Comment (agramfort) [inline mne/epochs.py]: approved

File: mne/epochs.py [hunk 2]
```diff
@@ -1411,17 +1411,17 @@ def __iter__(self):
...
         return_event_id : bool
-            If True, return both an epoch and and event_id.
+            If True, return both the epoch data and an event_id.
 
         Returns
         -------
-        epoch : instance of Epochs
-            The epoch.
+        epoch : numpy.ndarray
+            The epoch data (n_channels by n_times).
```
Comment (agramfort) [inline mne/epochs.py]: shape should be above

```
epoch : array, shape (n_channels, n_times)
     The epoch data
```

Comment (drammock) [inline mne/epochs.py]: fixed in 5d9dd60

**Comment by jasmainak** [comment]
Thanks for the quick fix @drammock !

**Pull request merged**